modules/billing-account/README.md

```diff
@@ -212,21 +213,63 @@ module "billing-account" {
 # tftest modules=1 resources=2 inventory=budget-monitoring-channel.yaml
 ```
 
+#### Budget factory
+
+This module also exposes a factory for billing budgets, that works in a similar way to factories in other modules: a specific folder is searched for YAML files, which contain one budget description per file. The file name is used to generate the key of the resulting map of budgets, which is merged with the one coming from the `budgets` variable. The YAML files support the same type of the `budgets` variable.
+
+```hcl
+module "billing-account" {
+  source = "./fabric/modules/billing-account"
+  id     = "012345-ABCDEF-012345"
+  budget_notification_channels = {
+    billing-default = {
+      project_id = "tf-playground-simple"
+      type       = "email"
+      labels = {
+        email_address = "gcp-billing-admins@example.com"
+      }
+    }
+  }
+}
```

Comment:
missing:
```
factory_configs = {
  budgets_folder = "data/billing-budgets"
}
```
?

Reply:
there's a default in the variable...

Reply:
meaning, the module tries the folder regardless of configuration; a filesystem lookup is a trivial penalty compared to all the rest that happens in a terraform plan/apply, IMHO :)

Reply:
![image](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/assets/66007/57ca29c0-5b03-4ab3-bb1c-6a1c58216b6a)

Reply:
the idea is that all factory-supporting modules will share the same variable, where `data_path` can be set to the same folder, and then one additional attribute points to the location of the factory files inside it

Reply:
:facepalm: I guess this might be worth pointing out somewhere in `README.md` that by default, the module will look for factory files in specific paths, or make `data_path` null by default and require setting it to enable this functionality.

Reply:
let's point out the default in the README, the split variable might also be confusing, but using a single variable would then make the type string (the folder path), and some factories might have more than one source of data / support more than one resource type

I might be falling in the premature optimization tunnel though :)

Reply:
It kind of feels like a bit like magic and explaining to users:
> don't create `data/` folder in your root as Fabric will peek there and pull stuff from there

Though I agree, that probability that someone hits the same naming pattern and DO NOT want to use this feature is like <1%.

Reply:
let me revert to a simpler thing, we will have time to make it more complex :)